ssl/ssl_ciph.c

```diff
@@ -519,22 +529,19 @@ int ssl_cipher_get_evp(SSL_CTX *ctx, const SSL_SESSION *s,
     if (c == NULL)
         return 0;
     if (comp != NULL) {
-        SSL_COMP ctmp;
 #ifndef OPENSSL_NO_COMP
-        if (!load_builtin_compressions()) {
-            /*
-             * Currently don't care, since a failure only means that
-             * ssl_comp_methods is NULL, which is perfectly OK
-             */
-        }
-#endif
+        SSL_COMP ctmp;
+
         *comp = NULL;
         ctmp.id = s->compress_meth;
-        if (ssl_comp_methods != NULL) {
+        if (load_builtin_compressions()
+            && CRYPTO_THREAD_write_lock(ssl_comp_methods_lock)) {
             i = sk_SSL_COMP_find(ssl_comp_methods, &ctmp);
```

Comment:
Worth a comment about how this could sort the stack?
Alternatively, force the sort as part of `load_building_compressions()` and make this a read lock.